docs/developing-operators.md

```diff
@@ -84,6 +84,75 @@ Now your first operator is ready and you can install it to cluster. You can do i
 
 For simplicity if you want to install what we created here without actually replicating it on your filesystem your can just clone KUDO repository and then run `kubectl kudo install ./config/samples/first-operator`.
 
+## Operator.yaml file
+
+This is the main piece of every operator. It consists of two main parts. First one defines metadata about your operator.
+
+```yaml
+name: operator
+description: my first super awesome operator
+version: "5.7"
+kudoVersion: ">= 0.2.0"
+kubeVersion: ">= 1.14"
+maintainers:
+  - Bob <bob@example.com>
+  - Alice <alice@example.com>
+url: https://github.com/myoperator/myoperator
+```
+
+Most of these are provided as a form of documentation. `kudoVersion` and `kubeVersion` use semver constraints to define minimal or maximal version of kubernetes or kudo that this operator supports. Under the hood, we use [this library](https://github.com/Masterminds/semver) to evaluate the constraints.
```

Comment:
IMO, I don't think we need to reference the library as it is an implementation detail - if the library has nuances that Kubernetes / semantic version doesn't have then that sounds like a bug.

Reply:
well the reasoning here was that I think semver constraints have no specifications, people implement it differently. Ideally I should probably explain what is the format that kudo expects, I made it easier for myself by linking the library :D I can change that... no strong opinion, just laziness :)

Reply:
Yeah by looking at their docs I would have to port over a lot of readme about how to define ranges etc. :/